pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -359,6 +382,41 @@ public synchronized void readMoreEntries() {
         }
     }
 
+    private boolean shouldPauseOnAckStatePersist(ReadType readType) {
+        // Allows new consumers to consume redelivered messages caused by the just-closed consumer.
+        if (readType != ReadType.Normal) {
+            return false;
+        }
+        if (!((PersistentTopic) subscription.getTopic()).isDispatcherPauseOnAckStatePersistentEnabled()) {
+            return false;
+        }
+        if (cursor == null) {
+            return true;
+        }
+
+        /**
+         * Double check for "Cursor data can be fully persistent".
+         * - Clear the marker that represent some acknowledgements were executed.
+         * - Check whether dispatching should be paused due to cursor data is too large to persistent.
+         * - If dispatching should be paused, but some acknowledgements have been executed, re-calculate the result.
+         * - Mark delivery was paused at least once.
+         */
+        // Clear the marker that represent some acknowledgements were executed.
+        blockSignatureOnCursorDataCanNotFullyPersist.clearMakerNewAcknowledged();
```

Comment:
Maybe you should move `blockSignatureOnCursorDataCanNotFullyPersist.clearMakerNewAcknowledged();` after `blockSignatureOnCursorDataCanNotFullyPersist.markPaused();`. That will trigger `readMoreEntriesAsync` more one time, but will not lose.

Reply:
I could not fully understand this comment. Could you explain in more detail?

Reply:
This is following the [comment](https://github.com/apache/pulsar/pull/21423/files/560944a7409df2cce497e8432da2b95b61d25325#diff-89265294ca675086422dde3f59e67ef298e7b7eea4d1dc8d28de7b9d60f92c0b). The original approach is fine.